test/index.ts

```diff
@@ -375,6 +384,22 @@ function testDeepOmit2() {
   type Test = Assert<IsExact<Result, Omitted>>;
 }
 
+function testDeepPick() {
+  type OptionalProperty = {
+    id: string;
+    age: number;
+    name?: string;
+  };
+  type Omitted = {
+    id: string;
+    name?: string;
+  };
+
+  type Result = DeepOmit<OptionalProperty, { age: never }>;
```

Comment:
shouldnt it be `DeepPick`?

Reply:
Yes is should...

Update: upon consideration I discarded additional `testDeepOmit2` and `test deepPick` functions altogether and integrated all checks in one function `testDeepOmitAndDeepPick()` to make everything more neat. 
Test for commented lines will fail in the moment, this is what will be fixed by next PR: https://github.com/krzkaczor/ts-essentials/issues/176
Or should we do it all at once?

Reply:
As for `true` instead of `never` , I thought along these lines too.
If we consider that Filter type may be used by `DeepPick` and `DeepOmit` simultaneously, `true` seems more adequate. But there's one implication that i found, is we rewrite the code using 'true', 'never' will also still work. So mixing both in Filter becomes possible. One also will be able to write filter types for DeepPick using `true` and filter types for DeepOmit using `never`, if that is his preference.

Reply:
Before we finally went with `never` I originally proposed the filter class to use `true`, so I'd have nothing against it, it still fits intuitively (you're selecting the properties to be ommited).
@krzkaczor?

Reply:
So I would propose on making a breaking change and then `DeepOmit` should use `false` and the new type, `DeepPick` should use `true`. Does it make sense? Or should we simply use just `true` in both cases?

Reply:
`true` in both cases, the whole point of the breaking change is to use the same type here